test/e2e_node/system/validators.go

```diff
@@ -54,6 +47,21 @@ func Validate(spec SysSpec, report Reporter) error {
 }
 
 // ValidateDefault uses all default validators to validate the system and writes to stdout.
-func ValidateDefault() error {
-	return Validate(DefaultSysSpec, DefaultReporter)
+func ValidateDefault(criEnabled bool) error {
+	// OS-level validators.
+	var osValidators = []Validator{
+		&OSValidator{Reporter: DefaultReporter},
+		&KernelValidator{Reporter: DefaultReporter},
+		&CgroupsValidator{Reporter: DefaultReporter},
+	}
+	// Docker-specific validators.
+	var dockerValidators = []Validator{
+		&DockerValidator{Reporter: DefaultReporter},
+	}
+
+	validators := osValidators
+	if !criEnabled {
+		validators = append(validators, dockerValidators...)
```

Comment:
I think we should run different validation for different runtimes, instead of just skipping it when cri is enabled.

Let's add a flag `container-runtime` instead of the `cri-enabled` flag.